common/djangoapps/util/testing.py

```diff
@@ -55,3 +57,36 @@ def setUp(self, *args, **kwargs):
 
         self._reset_urls(urlconf_modules)
         self.addCleanup(lambda: self._reset_urls(urlconf_modules))
+
+
+class EventTestMixin(object):
```

Comment:
Good question. `EventTestMixin` patches the tracking library at the unit test level, while `EventsTestMixin` watches the test database's "events" collection, since it does not have access to the django server. I'm not really sure what should be shared between these two classes other than an abstract set of methods. Even then, it seems to me like the use cases of each class could change, and I'm not sure what we'd win by sharing code.

I do agree that the naming is confusing. What if we were to change the names to make things more clear? How about `EventPatcherMixin` and `EventWatcherMixin` to convey the difference between the two classes?

I'm going to hold off on this refactor until I figure out why events are not being fired (and sometimes fired twice) from the account settings page.